lib/base/tlsutility.cpp

```diff
@@ -72,18 +73,18 @@ void InitializeOpenSSL()
 	l_SSLInitialized = true;
 }
 
-static void InitSslContext(const Shared<boost::asio::ssl::context>::Ptr& context, const String& pubkey, const String& privkey, const String& cakey)
+static void InitSslContext(const Shared<TlsContext>::Ptr& context, const String& pubkey, const String& privkey, const String& cakey)
 {
 	char errbuf[256];
 
 	// Enforce TLS v1.2 as minimum
 	context->set_options(
-		boost::asio::ssl::context::default_workarounds |
-		boost::asio::ssl::context::no_compression |
```

Comment:
MS seems to have disabled compression by default, so we don't need/have an equivalent here.

* https://learn.microsoft.com/en-us/aspnet/core/performance/response-compression?view=aspnetcore-5.0#compression-with-secure-protocol
* https://learn.microsoft.com/en-us/answers/questions/1072954/disable-tls-compression-on-windows-server
* https://chatgpt.com/share/3ba78667-0741-43e3-9eb1-ce77a63cd05b

Reply:
GnuTLS 3.6.0 and even OpenSSL 1.1.1 disabled compression, too.

* https://gnutls.org/manual/gnutls.html#Compression-algorithms-and-the-record-layer
* https://wiki.openssl.org/index.php/List_of_SSL_OP_Flags